test/e2e/admin/admin-checkout.test.ts

```diff
@@ -0,0 +1,317 @@
+import * as firebase from 'firebase';
+import { slow, suite, test } from 'mocha-typescript';
+import squel from 'squel';
+import { CheckoutItems } from '../../../src/models/checkout-items/checkout-items';
+import { CheckoutObject } from '../../../src/models/checkout-object/checkout-object';
+import { IntegrationTest } from '../integration-test';
+import { TestData } from '../test-data';
+
+let listener: firebase.Unsubscribe;
+let firebaseUser: firebase.User;
+
+function login(email: string, password: string): Promise<firebase.User> {
+  if (firebaseUser) {
+    return new Promise(resolve => resolve(firebaseUser));
+  }
+  return new Promise((resolve, reject) => {
+    firebase.auth()
+      .signInWithEmailAndPassword(email, password)
+      .catch(err => reject(err));
+    listener = firebase.auth()
+      .onAuthStateChanged((user) => {
+        if (user) {
+          firebaseUser = user;
+          resolve(user);
+        }
+      });
+  });
+}
+
+function loginAdmin() {
+  return login('admin@email.com', 'password');
+}
+
+@suite('INTEGRATION TEST: Admin Checkout')
+class AdminCheckoutIntegrationTest extends IntegrationTest {
```

Comment:
Does this delete the items that are inserted after? If not, will the lack of idempotence be a problem? Ensure there are no unique constraints that become invalid after one instance is inserted.

Reply:
The after function in IntegrationTest calls the TestData.cleanup() after every test which will wipe all tables. So for the most part, all the sets of tests can be run individually, but one individual test may depend on others in the same set for example when doing deletions.